From 30cefc69e39b6fc74e7104162a74a33a7e72f7d1 Mon Sep 17 00:00:00 2001
From: David Steele <dsteele@gmail.com>
Date: Mon, 7 Oct 2013 18:29:45 -0400
Subject: [PATCH 14/20] Unhandled exception in XMPP recv.

The following was in the user.log.

Traceback (most recent call last):
012  File "/usr/share/cloudprint/cloudprint/cloudprint.py", line 373, in process_jobs
012    xmpp_conn.await_notification(sleeptime)
012  File "/usr/share/cloudprint/cloudprint/xmpp.py", line 189, in await_notification
012    self._read_socket()
012  File "/usr/share/cloudprint/cloudprint/xmpp.py", line 55, in _read_socket
012    data = self._wrappedsock.recv(1024)
012  File "/usr/lib/python2.7/ssl.py", line 305, in recv
012    return self.read(buflen)
012  File "/usr/lib/python2.7/ssl.py", line 224, in read
012    return self._sslobj.read(len)
012error: [Errno 104] Connection reset by peer

Assume that this is because the exception is not of type 'Exception',
and change the except in process_jobs to pure except:.
(cherry picked from commit a7cdfdb1fca55cf8d83698f49945f60aef5f9adc)
---
 cloudprint/cloudprint.py | 2 +-
 cloudprint/xmpp.py       | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/cloudprint/cloudprint.py b/cloudprint/cloudprint.py
index 370fa89..43b3b35 100755
--- a/cloudprint/cloudprint.py
+++ b/cloudprint/cloudprint.py
@@ -372,7 +372,7 @@ def process_jobs(cups_connection, cpp, printers):
 
             xmpp_conn.await_notification(sleeptime)
 
-        except Exception:
+        except:
             global FAIL_RETRY
             LOGGER.exception('ERROR: Could not Connect to Cloud Service. Will Try again in %d Seconds') % FAIL_RETRY
             time.sleep(FAIL_RETRY)
diff --git a/cloudprint/xmpp.py b/cloudprint/xmpp.py
index a1b83cf..9667c35 100644
--- a/cloudprint/xmpp.py
+++ b/cloudprint/xmpp.py
@@ -56,7 +56,7 @@ class XmppConnection(object):
             if data is None or len(data) == 0:
                 # socket closed
                 raise Exception("xmpp socket closed")
-        except Exception:
+        except:
             self._connected = False
             raise
 
@@ -192,7 +192,7 @@ class XmppConnection(object):
                     LOGGER.warn("Error in xmpp connection")
                     raise Exception("xmpp connection errror")
 
-            except Exception:
+            except:
                 self.close()
                 raise
 
-- 
1.8.4.rc3

